protocols/discord.py

```diff
@@ -500,6 +501,26 @@ def message(self, source, target, text, notice=False):
             self.virtual_parent.message_queue.put_nowait(message_data)
             return
 
+        if self.virtual_parent.serverdata.get('webhooks', False) and self.is_channel(target):
+            try:
+                webhook = self._get_or_create_webhook(self.channels[target])
+            except APIException:
+                log.debug('(%s) _get_or_create_webhook: could not get or create webhook for channel %s. Falling back to standard Clientbot behavior', self.name, target, exc_info=True)
+                webhook = None
+
+            if webhook:
+                try:
+                    remotenet, remoteuser = self.users[source].remote
+                    message_data['webhook'] = webhook
+                    message_data.update(self._get_user_webhook_data(remoteuser, remotenet))
+                    message_data['text'] = I2DFormatter().format(text)
```

Comment:
I think we should pull this call earlier in the file, since it's effectively being duplicated.

Reply:
One of the main reasons I don't have the formatter call earlier on is so the message won't get formatted before emitting the `CLIENTBOT_MESSAGE` hook. If we wanna reduce duplication, I can move the formatter into the message builder thread, so it only gets applied before a message gets flushed.

Reply:
That sounds like a good idea